### PR TITLE
Console - Lookups view

### PR DESCRIPTION
**Pull request by shuqi7**
Refers to Druid lookups: http://druid.io/docs/latest/querying/lookups.html

- Lookups page to allow users to view existing lookups, add new lookups, and edit lookups
image
- Allow users to use ISO string as current version, and will pre-fill current ISO as version when users add new lookups
image
- Spec refers to the JSON object in `lookupExtractorFactory`
- If the user has NEVER configured lookups before, he must initialize the configuration first
image

### CODE REVIEW

File: web-console/src/views/lookups-view.tsx [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+  componentWillUnmount(): void {
+    this.lookupsGetQueryManager.terminate();
+    this.lookupDeleteQueryManager.terminate();
+  }
+
+  private async openLookupEditDialog(tier:string, name: string) {
+    let endpoint = `/druid/coordinator/v1/lookups/config`;
+    if (tier !== "" && name !== "") endpoint = `${endpoint}/${tier}/${name}`;
+    const resp = await axios.get(endpoint);
```
Comment (vogievetsky) [inline web-console/src/views/lookups-view.tsx]: why is this needed, don't you already have this data?

File: web-console/src/views/lookups-view.tsx [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+          const lookupIds = lookupData[tier];
+          Object.keys(lookupIds).map((id: string) => {
+            lookupEntries.push({tier: tier, id: id, type: lookupData[tier][id].lookupExtractorFactory.type});
+          })
+        })
+        return lookupEntries;
+      },
+      onStateChange: async ({ result, loading, error }) => {
+        if (result === null || error !== null) {
+          await axios.post(`/druid/coordinator/v1/lookups/config`, {})
```
Comment (vogievetsky) [inline web-console/src/views/lookups-view.tsx]: ooo this is rough, the onStateChange should not be `async` furthermore, it should not be doing the lookup initialization on the sly. having un-initialiaed lookups should be a proper state of the application and there should be a button to initialize lookups

File: web-console/src/views/lookups-view.tsx [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+            .then(() => this.lookupsGetQueryManager.rerunLastQuery())
+            .catch(e => console.error(e));
```
Comment (vogievetsky) [inline web-console/src/views/lookups-view.tsx]: Never use console error. pop up the toast instead

File: web-console/src/console-application.tsx [hunk 4]
```diff
@@ -32,6 +32,7 @@ import { ServersView } from './views/servers-view';
 import { TasksView } from './views/tasks-view';
 import { SqlView } from './views/sql-view';
 import "./console-application.scss";
+import {LookupsView} from "./views/lookups-view";
```
Comment (vogievetsky) [inline web-console/src/console-application.tsx]: please fix ordering (next to the other views)

File: web-console/src/views/lookups-view.tsx [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+          },
+          {
+            Header: "Tier",
+            accessor: "tier",
+            filterable: true,
+            Cell: row => row.value
+          },
+          {
+            Header: "Type",
+            accessor: "spec",
```
Comment (vogievetsky) [inline web-console/src/views/lookups-view.tsx]: you are misusing accessor here, should be a function

**Comment by vogievetsky** [comment]
Looking over this now

**Comment by shuqi7** [comment]
Made a PR for Apache Druid
